tools/ac_harness/shared_memory.py

```
@@ -0,0 +1,454 @@
+            if section is not None:
+                section.close()
+
+    def __enter__(self) -> SharedMemoryReader:
+        return self
+
+    def __exit__(self, *exc: object) -> None:
+        self.close()
+
+
+def _open_reader_with_retry(  # pragma: no cover - rig-only live-probe helper
+    *, attempts: int, interval: float, clock: Callable[[], float], sleep: Callable[[float], None]
```

Ruff (0.15.15)

[warning] Unused function argument: `clock`

(ARG001)